### ISSUE TITLE
Determine deathrattle orders with cleave

### ISSUE THREAD

**Issue by jeremysalwen** [JDBumgardner/stone_ground_hearth_battles#8]
We need to know what order deathrattles are resolved if two deathrattle minions are killed simultaneously with cleave. We also need to know if minions can be revived from deathrattle buffs in this case.

It would be especially interesting to see the ordering of resolution if a kaboom bot is killed, and triggers a deathrattle on the opposing team.

Experiments which result in different *outcomes* at the end of the attack are preferred over experiments where the order of animations is merely different.

**Comment by jeremysalwen** [comment]
For context @jazzcs  says

> In BG it's a little complicated
> there's an order to deathrattles
> which starts as something like left to right on board
> but if new minions spawn
> they're put at the end of the order
> it's something I don't have an exact grasp on, and it would be good to test in client at some point
> (end of the order, even if they're earlier left to right)

**Comment by JDBumgardner** [comment]
So, I propose that to implement this ordering we could make a priority queue, give every card a priority based on its position / when its created, and then just have like a while loop event trigger resolution bit inside combat.

There are two things that it occurs to me we have to figure out still. 

1. How is priority handled between the two boards when both players have event triggers that need to be resolved? Are the cards interleaved in terms of their priority or is there some active player versus non active player handling. 

2. For deathrattles the card that is dying is also the card whose event is being triggered, however, there are triggers that are broadcast by one card and handled by another. In this case, which card's priority is used? Maybe this could be thought about as in is the broadcasting of the event put into the queue or is the effect to be resolved put into the queue?

I'm down to do some testing in the client

**Comment by ethansaxenian** [comment]
Here's an interesting experiment:

Board 1: 
    3+ power cleave minion / rando / rando / rando / rando

Board 2: 
    1-toughness dude / Selfless Hero / Unstable Ghoul / rando (should die to cleave)


Does 1-toughness dude survive? If deathrattle ordering is left-to-right, it should. Can also try swapping the minions on each side of the ghoul.

**Comment by jeremysalwen** [comment]
That's a good idea, we should definitely run it multiple times though because it seems like there may be hidden variables.

**Comment by ethansaxenian** [comment]
Yeah I think there are definitely ways to set this up so it's a tie every round. The hero Deathwing is the first thought that pops into my mind.
